src/scanner.rs

```diff
@@ -161,8 +162,26 @@ impl BundleRepo {
                 .string_by_key(GIT_REMOTE_ORIGIN_URL_CONFIG)
                 .map(|s| s.to_string());
             let mut git_head = git_repo.head()?;
+
+            let mut git_head_branch = git_head.referent_name().map(|s| s.as_bstr().to_string());
+            if git_head_branch.is_none() {
+                let git_head_id = git_head.id().unwrap();
+                let reference = git_repo.references().unwrap();
+                reference.remote_branches()?.for_each(|r| match r {
+                    Ok(r) => {
+                        let target = r.target();
+                        let id = target.try_id();
+                        if id.is_some() && id.unwrap().to_string() == git_head_id.to_string() {
+                            git_head_branch =
+                                Some(r.name().to_path().to_str().unwrap().to_string());
+                        };
```

Comment:
I'm sure there's a more elegant solution here. If there is, any pointers would be greatly appreciated.